packages/core/src/llm/model/router.ts

```diff
@@ -71,7 +76,9 @@ export class ModelRouterLanguageModel implements LanguageModelV2 {
     };
 
     // Resolve gateway once using the normalized ID
-    this.gateway = findGatewayForModel(normalizedConfig.id, gateways);
+    // Use custom gateways if provided, otherwise use default gateways
+    const gatewaysToUse = customGateways && customGateways.length > 0 ? customGateways : defaultGateways;
+    this.gateway = findGatewayForModel(normalizedConfig.id, gatewaysToUse);
```

Comment:
@DanielSLew This would prevent the default gateways from being resolved if there is a custom gateway added right? Should we just pass in merged gateways to `findGatewayForModel`?

```suggestion
    this.gateway = findGatewayForModel(normalizedConfig.id, [...customGateways, ...defaultGateways]);
```